prov/verbs/src/verbs_info.c

```diff
@@ -1325,7 +1325,9 @@ int vrb_init_info(const struct fi_info **all_infos)
 
 	/* note we are possibly discarding ENOMEM */
 	ret = *all_infos ? 0 : ret;
-
+	for (i = 0; i < num_devices; i++) {
+		ibv_close_device(ctx_list[i]);
+	}
```

Comment:
This is going under the librdmacm library interface to close devices that that library expects to have available.  rdma_free_devices()  keeps the devices open intentionally.

The Intel CI failures are segfaults in verbs tests.

Reply:
@shefty, could you please take a look? Added a check, instead of closing the device